Formula/infer.rb

```diff
@@ -69,36 +57,11 @@ def install
     # Let's try build clang faster
     ENV["JOBS"] = ENV.make_jobs.to_s
 
-    ENV["CLANG_CMAKE_ARGS"] = "-DLLVM_OCAML_INSTALL_PATH=#{`opam var lib`.chomp}/ocaml"
-
     # Release build
     touch ".release"
 
-    # Pin updated dependencies which are required to build on brew ocaml
-    # Remove from this when Infer updates their opam.locked to use at least these versions
-    pinned_deps = {
-      "mlgmpidl"  => "1.2.12",
-      "octavius"  => "1.2.1",
-      "parmap"    => "1.0-rc11",
-      "ppx_tools" => "5.3+4.08.0",
-    }
-    pinned_deps.each { |dep, ver| system "opam", "pin", "add", dep, ver, "--locked" }
-
-    # Unfortunately, opam can't cope if a system ocaml-num happens to be installed.
-    # Instead, we depend on Homebrew's ocaml-num and fool opam into using it.
-    # https://github.com/ocaml/opam-repository/issues/14646
-    system "opam", "pin", "add", "ocamlfind", Formula["ocaml-findlib"].version.to_s, "--locked", "--fake"
-    system "opam", "pin", "add", "num", Formula["ocaml-num"].version.to_s, "--locked", "--fake"
-
-    # Relax the dependency lock on a specific ocaml
-    # Also ignore anything we pinned above
-    ENV["OPAMIGNORECONSTRAINTS"] = "ocaml,ocamlfind,num,#{pinned_deps.keys.join(",")}"
-
-    # Remove ocaml-variants dependency (we won't be using it)
-    inreplace "opam.locked", /^ +"ocaml-variants" \{= ".*?"\}$\n/, ""
-
-    system "opam", "exec", "--", "./build-infer.sh", "all", "--yes", "--user-opam-switch"
-    system "opam", "exec", "--", "make", "install-with-libs"
+    system "./build-infer.sh", "all", "--yes"
+    system "make", "install-with-libs"
```

Comment:
Does this download any dependencies?

Reply:
The first of these two steps downloads dependencies from opam and llvm:
- opam for OCaml libraries needed to build infer
- llvm because infer ships with a clang plugin that only works against a particular version of clang

Reply:
Homebrew formulae need to work with the Homebrew versions of dependencies. That's what it's a package manager for.

Reply:
These dependencies do not affect the rest of the system in any way, even once infer is installed. Opam dependencies are moreover only needed to build infer and almost none of them are available as Homebrew packages.

Reply:
Managing the opam/ocaml dependencies via Homebrew would probably require a resource-staging approach with a tool similar to [homebrew-pypi-poet](https://github.com/tdsmith/homebrew-pypi-poet) (for pip/python packages)